parl/framework/agent.py

```diff
@@ -0,0 +1,95 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta, abstractmethod
+from multiprocessing import Process, Value
+from py_simulator import Simulator
+from parl.common.replay_buffer import Experience
+from parl.common.logging import GameLogEntry
+
+
+class AgentBase(Process):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, game_name, game_options, num_games):
+        super(AgentBase, self).__init__()
+        self.id = -1  # isolated agent, not recognized by the framework yet
+        self.game_name = game_name
+        self.game = Simulator.create(game_name, game_options)
+        self.num_games = num_games
+        self.helpers = {}
+        self.log_q = []
+        self.exit_flag = Value('i', 0)
+
+    def add_helper(self, helper):
+        self.helpers[helper.name] = helper
+
+    def set_log_queue(log_q):
+        self.log_q = log_q
+
+    @abstractmethod
+    def _run_one_episode(self):
+        pass
+
+    @abstractmethod
+    def _game_over(self, game_status):
+        pass
+
+    @abstractmethod
+    def _game_success(self, game_status):
+        pass
+
+    def run(self):
+        for _ in range(self.num_games):
+            self._run_one_episode()
+
+
+class SimpleRLAgent(AgentBase):
```

Comment:
I am not quite sure which classes are example implementations and which are fixed base classes. You probably should put all example classes somewhere else.

Reply:
Sounds good, but only put agents that are representative inside that folder